src/input/parse_json.rs

```diff
@@ -1,113 +1,78 @@
 use std::fmt;
 
 use num_bigint::BigInt;
-use pyo3::prelude::*;
-use pyo3::types::{PyDict, PyList};
 use serde::de::{Deserialize, DeserializeSeed, Error as SerdeError, MapAccess, SeqAccess, Visitor};
 
+use crate::data_value::DataValue;
 use crate::lazy_index_map::LazyIndexMap;
 
-/// similar to serde `Value` but with int and float split
-#[derive(Clone, Debug)]
-pub enum JsonInput {
-    Null,
-    Bool(bool),
-    Int(i64),
-    BigInt(BigInt),
-    Uint(u64),
-    Float(f64),
-    String(String),
-    Array(JsonArray),
-    Object(JsonObject),
-}
+pub type JsonInput = DataValue;
```

Comment:
do we still need `JsonInput` at all?

Reply:
No not really, this was just to reduce code churn.